Add ES6 transpile option to the source sending.

knightburton opened the pull request:
With this feature the users can able to write ES6 code in the editor,
with some limitation because of the enviroment.
- Add a new  ES6 transpile setting to the settings menu.
- Add a new getValue function to the settings module. This function can
  return with the value of a single setting.
- Add a new Transpiler module to the project. This module handles the
  source code transformation and source map to achieve our goals.
- Wrap the transpile related codes inside an if to avoid
  the unnecessary function calls if the transpile option is turned off.
- Modify the source line related actions and information providers.
  If the transpile is enabled we have to use the source map to
  show the proper state of the debugging process while the user
  see the ES6 source in the editor.

IoT.jsCode-DCO-1.0-Signed-off-by: Imre Kiss kissi.szeged@partner.samsung.com

daeyeon commented on src/app/transpiler.js:
```diff
@@ -0,0 +1,107 @@
+  transformToES5(filename, source) {
+    let name = filename.split('.')[0];
+
+    transpiled[name] = Babel.transform(source, {
```
Apparently, an error occurs during transpiling sources, app can stop. An appropriate exception handling seems needed.

knightburton replied:
Can you send me that what you have tested?

daeyeon replied:
You may observe that in the following sequence.

i) iotjs debugger waits a source.
ii) enable transpiling to ES5 and create a new file.
iii) put the below code into the editor panel. The spaces between characters seem needed.

```javascript
t e s t
```
iv) send the source to the server.
v) iotjscode crashes.

I hope this can help you.

knightburton replied:
Yes, thank you, I see the problem now. I will figure it out something :)

knightburton commented:
@daeyeon I've updated the PR, please take a look at it. I've modified the logger module a bit to show proper message to the user in case of transpile error.
